packages/react-components/react-utilities/etc/react-utilities.api.md

```diff
@@ -211,14 +217,47 @@ export type Slot<Type extends keyof JSX.IntrinsicElements | React_2.ComponentTyp
     } & WithSlotRenderFunction<IntrinsicElementProps<As>>;
 }[AlternateAs] | null : 'Error: First parameter to Slot must not be not a union of types. See documentation of Slot type.';
 
+// @public
+export function slot<Props extends UnknownSlotProps>(value: Props | SlotComponent<Props> | SlotShorthandValue | undefined, options: {
+    required: true;
+} & SlotOptions<Props>): SlotComponent<Props>;
+
+// @public (undocumented)
+export function slot<Props extends UnknownSlotProps>(value: Props | SlotComponent<Props> | SlotShorthandValue | undefined | null, options: {
+    required?: boolean;
+} & SlotOptions<Props>): SlotComponent<Props> | undefined;
+
+// @public (undocumented)
+export function slot<Props extends UnknownSlotProps>(value: SlotComponent<Props>, options?: {
+    required: true;
+} & Partial<SlotOptions<Props>>): SlotComponent<Props>;
+
 // @internal
-export const SLOT_RENDER_FUNCTION_SYMBOL: unique symbol;
+export const SLOT_COMPONENT_METADATA_SYMBOL: unique symbol;
```

Comment:
nit: Technically it's a breaking change

Reply:
should we keep it and deprecate it then?! 🙈

Reply:
Yes, let's keep it deprecated.